thunder/transforms/quantization.py

```diff
@@ -294,3 +295,214 @@ def transform_traces_pre_prologue(self, prologue_trace, computation_trace, epilo
+                    new_compute_inputs.append(proxy_loraB)
+                    # this is not good, because we will have several traces...
+                    additional_proxies[n_loraA] = proxy_loraA
+                    additional_proxies[n_loraB] = proxy_loraB
+
+        prologue_trace.bound_symbols[-1:-1] = new_bsyms
+
+        computation_proxy_map = {
+            csym.name: dict(
+                shape=psym.shape,
+                dtype=psym.dtype,
+            )
+            for psym, csym in zip(prologue_trace.bound_symbols[-1].args[0][0], computation_trace.args)
+            if psym.shape != csym.shape or psym.dtype != csym.dtype
+        }
+
+        new_computation_trace = trace_with_replaced_proxy_metadata(computation_trace, computation_proxy_map)
+        bound_symbols = new_computation_trace.bound_symbols
+        new_computation_trace.bound_symbols = []
+
+        new_computation_trace.args = (*new_computation_trace.args, *new_compute_inputs)
+        new_computation_trace._siginfo.args = [(a.name, None) for a in new_computation_trace.args]
+
+        with tracectx(new_computation_trace):
+            new_bindings = [
+                thunder.core.prims.unpack_trivial.bind(i, output=i, name=i.name) for i in new_compute_inputs
+            ]
+
+        for idx, bsym in enumerate(bound_symbols):
+            if bsym.sym != prims.unpack_trivial:
+                break
+            new_computation_trace.bound_symbols.append(bsym.from_bsym())
+        new_computation_trace.bound_symbols += new_bindings
+
+        for bsym in bound_symbols[idx:]:
+            if bsym.sym == thunder.torch.linear:
+                n = lora_linear_proxies[bsym.args[1].name]
+                with thunder.core.trace.tracectx(computation_trace):
+                    loraA_transpose_meta = prims.transpose.meta(additional_proxies[f"{n}.loraA"], (1, 0))
+                    loraA_bsym = prims.transpose.bind(
+                        additional_proxies[f"{n}.loraA"],
+                        (1, 0),
+                        output=(loraA_transpose_meta),
+                    )
+                    loraB_transpose_meta = prims.transpose.meta(additional_proxies[f"{n}.loraB"], (1, 0))
+                    loraB_bsym = prims.transpose.bind(
+                        additional_proxies[f"{n}.loraB"],
+                        (1, 0),
+                        output=(loraB_transpose_meta),
+                    )
+                    lora_meta_ = prims.matmul.meta(bsym.args[0], loraA_bsym.output)
+                    lora_bsym_ = prims.matmul.bind(bsym.args[0], loraA_bsym.output, output=lora_meta_)
+                    lora_meta = prims.matmul.meta(lora_bsym_.output, loraB_bsym.output)
+                    lora_bsym = prims.matmul.bind(lora_bsym_.output, loraB_bsym.output, output=lora_meta)
+
+                    # Is there a better way?
+                    original_weight_meta = prims.linear.meta(*bsym.args[:3])
```

Comment:
this is OK, I guess but it would be nicer to create a new `TensorProxy(like=bsym.output)` and then calling `bsym.from_bsym(output=...)` and append that into the trace to preserve the metadata (like source line info).

Reply:
Let me do the following:
```
                    original_weight = bsym.from_bsym(
                        sym=prims.linear,
                        args=(bsym.args[:3]),
                    )
                new_computation_trace.bound_symbols.append(original_weight)
```